src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/ClusterDimensionalSummary.java

```diff
@@ -241,8 +259,9 @@ public static ClusterDimensionalSummary build(Record record, DSLContext context)
                 Integer.class);
 
         ClusterDimensionalSummary summary =
-                new ClusterDimensionalSummary(TemperatureVector.Dimension.valueOf(dimensionName));
-        summary.setTotalUsage(total);
+                new ClusterDimensionalSummary(TemperatureDimension.valueOf(dimensionName));
```

Comment:
Since this code would be running on the Nodes itself and we would be fetching this from Nodestats it is highly unlikely the metric would be populated as zero. The only extreme case that is possible is when the metric is not populated and default value etc is picked up. Anyways it is always good to have the check. Added.